### DISCUSSION TITLE
[Q] How to add extra container and modify creation

### DISCUSSION THREAD

**Discussion by ParadiseFallen** [mobxjs/mobx-state-tree#1979] [Q&A]
Hi. I have a questuion about extending mst.

Im using kinda MVVM with react. so i need to pass dependencies to my models.
And i dont want to use env (has no strong type)

So im woundering how to create `.dependencies()` method for types.model builder and add types from dependencies to required params on `create` method

**Comment by EmilTholin** [comment]
Hi @ParadiseFallen!

You could e.g. create your own `views` extension that supplies a typed version of the environment:

```ts
import { getEnv, IStateTreeNode, types } from "mobx-state-tree";

type Env = {
  baz: string;
};

const withEnv = (self: IStateTreeNode) => ({
  views: {
    get env() {
      return getEnv<Env>(self);
    }
  }
});

const Model = types
  .model({
    foo: "bar"
  })
  .extend(withEnv)
  .actions((self) => ({
    afterCreate() {
      console.log(self.env.baz);
    }
  }));

const model = Model.create({}, { baz: "qux" });
```
